### PR TITLE
Port to ESLint

### PR DESCRIPTION
**Pull request by danni**
Replaces jscs and JSXHint. Includes React ESLint plugin. Lots of style fixes from porting the style guide.

Adds a few more style checks that are nice to have.

### CODE REVIEW

File: src/components/ServicePane.js [hunk 1]
```diff
@@ -52,8 +36,31 @@ export default class ServicePane extends React.Component {
...
+
+    async getSiblingServices(): Promise<void> {
+        var response = await this.props.service.getSiblingServices();
+
+        this.setState({siblings: response.objects});
+    }
+
     render(): ReactElement {
         var object = this.props.service;
+        var filteredPhoneKinds = new Set(["fax", "tty"]);
```
Comment (DanielHeath) [inline src/components/ServicePane.js]: Is this a logic change?

File: src/components/CollapsedOpeningTimes.js [hunk 2]
```diff
@@ -47,20 +45,20 @@ export default class CollapsedOpeningTimes extends React.Component {
...
-        },{
+        }, {
             day: "Thursday",
             open: "00:00:00",
             close: "23:00:00",
-        },]),
+        }]),
     };
 
     render(): React.Element {
```
Comment (pfctdayelise) [inline src/components/CollapsedOpeningTimes.js]: Not part of this change but I thought daniel recently updated all these to be ReactElement ?

File: src/components/ServicePane.js (hunk 1 above)
Comment (danni) [inline src/components/ServicePane.js]: Merge glitch.

**Comment by DanielHeath** [comment]
Whew, what a diff.

File: src/pages/StyleGuideList.js [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
-import React from 'react';
-import Router from 'react-router';
-import components from '../components';
+import React from "react";
+import Router from "react-router";
+import components from "../components";
 
 export default class StyleGuideList extends React.Component {
-    li(componentName: string): ReactElement {
+    render(): ReactElement {
```
Comment (pfctdayelise) [inline src/pages/StyleGuideList.js:8]: There's no reference to li anymore?

Comment (DanielHeath) [inline src/pages/StyleGuideList.js:8]: `li` was renamed to `renderItem`, presumably to satisfy the rule about how long identifiers should be.

**Comment by DanielHeath** [comment]
++

File: src/pages/StyleGuideList.js (hunk 3 above)
Comment (danni) [inline src/pages/StyleGuideList.js:8]: Indeed.

Comment (pfctdayelise) [inline src/pages/StyleGuideList.js:8]: Oh right. I thought there was an actual li somewhere because it's in a ul but now I see there never was, at least in this file.

**Comment by pfctdayelise** [comment]
Seems good
